--- perdition-1.11/perdition/db/nis/Makefile.am.orig	Mon Jan  6 03:11:05 2003
+++ perdition-1.11/perdition/db/nis/Makefile.am	Tue Aug 12 12:21:38 2003
@@ -32,7 +32,7 @@
 
 libperditiondb_nis_la_LDFLAGS    = -version-info 0:0:0
 
-libperditiondb_nis_la_LIBADD     = @nsl_lib@ -lvanessa_logger
+libperditiondb_nis_la_LIBADD     = @nsl_lib@ -lvanessa_logger @yp_lib@
 
 INCLUDES=\
 -I$(top_srcdir)/ \
--- perdition-1.11/configure.ac	Fri Jun  6 05:56:52 2003
+++ perdition-1.18/configure.ac	Tue Aug 12 12:22:01 2010
@@ -39,9 +39,9 @@
 
 # Checks for libraries.
 AC_CHECK_LIB(dl, dlopen,  [ dl_lib="-ldl" ], [ dl_lib="" ])
-AC_CHECK_LIB(nsl, yp_match, true, true)
-AC_CHECK_LIB(socket, socket, [ socket_lib="-lsocket" ], [ socket_lib="" ])
-AC_CHECK_LIB(nsl, gethostbyname, [ nsl_lib="-lnsl" ], [ nsl_lib="" ])
+AC_CHECK_LIB(nsl, yp_match, [ yp_lib="-lnsl" ], [ yp_lib="" ])
+AC_CHECK_FUNC(socket, [ socket_lib="" ], [AC_CHECK_LIB(socket, socket, [ socket_lib="-lsocket" ], [ socket_lib="" ])])
+AC_CHECK_FUNC(gethostbyname, [ nsl_lib="" ], [AC_CHECK_LIB(nsl, gethostbyname, [ nsl_lib="-lnsl" ], [ nsl_lib="" ])])
 AC_CHECK_LIB(crypt, crypt, [ crypt_lib="-lcrypt" ], [ crypt_lib="" ])
 AC_CHECK_LIB(
   popt,
@@ -1124,6 +1124,7 @@
 AC_SUBST(socket_lib)
 AC_SUBST(dl_lib)
 AC_SUBST(nsl_lib)
+AC_SUBST(yp_lib)
 AC_SUBST(resolv_lib)
 AC_SUBST(crypt_lib)
 AC_SUBST(pam_dir)
