--- perdition-1.17/perdition/db/daemon/Makefile.am~	2007-02-21 22:08:56.293320000 +0000
+++ perdition-1.17/perdition/db/daemon/Makefile.am	2007-02-21 22:08:50.576180000 +0000
@@ -36,9 +36,11 @@
 
 noinst_PROGRAMS = client server
 
+client_CFLAGS = $(AM_CFLAGS)
 client_SOURCES = client.c 
 client_LDADD = $(COMMON_LIBRARY_ADD)
 
+server_CFLAGS = $(AM_CFLAGS)
 server_SOURCES = server.c
 server_LDADD = $(COMMON_LIBRARY_ADD)
 
